ops/framework.py

```diff
@@ -827,6 +831,32 @@ def _next_event_key(self) -> str:
         self._stored['event_count'] += 1
         return str(self._stored['event_count'])
 
+    def _event_is_in_storage(
+        self, observer_path: str, method_name: str, event_path: str, event_data: Dict[str, Any]
+    ) -> bool:
+        """Check if there is already a notice with the same snapshot in the storage."""
+        # Check all the notices to see if there is one that is the same other
+        # than the event ID.
+        for (
+            existing_event_path,
+            existing_observer_path,
+            existing_method_name,
+        ) in self._storage.notices():
+            if (
+                existing_observer_path != observer_path
+                or existing_method_name != method_name
+                or existing_event_path.split('[')[0] != event_path.split('[')[0]
```

Comment:
Looking through `class Handle`, it seems that not all paths have a "key" index.

Reply:
That's true, but all events (from `BoundEvent`'s `emit`) do have one, and that's the only type of object that has its path in the `notices` table.

If the `event_path` somehow was missing the key, then the code would also still work, because it's getting the 0th element so it will still be checking the full path - there just won't be a `[1:]` that gets discarded.

Reply:
This assumption/logic deserves a comment, or maybe a unit test.
I'd be lost if I tried to eyeball this without your comment.